Allow Plug.Exception implementations to not be included

alvinlindstam opened the pull request:
This will allow users to write alternate implementations or
treat them as unhandled exceptions.

Does not change behaviour unless explicitly configured to
do so.

Fixes https://github.com/phoenixframework/phoenix_ecto/issues/86

josevalim commented on README.md:
````diff
@@ -108,6 +108,15 @@ Thanks to Elixir protocols, the integration between Phoenix and Ecto is simply a
+The `Plug.Exception` implementations for Ecto exceptions may be disabled by including the error in the mix configuration.
+
+```elixir
+config :ecto_phoenix,
````
`:phoenix_ecto`

josevalim commented on README.md:
````diff
@@ -108,6 +108,15 @@ Thanks to Elixir protocols, the integration between Phoenix and Ecto is simply a
+
+```elixir
+config :ecto_phoenix,
+  exclude_ecto_excaptions: [Ecto.NoResultsError]
````
excaptions -> exceptions

josevalim replied:
Actually, what if we call the key "exclude_ecto_exceptions_from_plug"?

josevalim commented on lib/phoenix_ecto/plug.ex:
```diff
@@ -1,15 +1,14 @@
-defimpl Plug.Exception, for: Ecto.InvalidChangesetError do
-  def status(_), do: 422
+for {exception, status_code} <- errors do
+  unless exception in Application.get_env(:phoenix_ecto, :exclude_ecto_excaptions, []) do
```
You should read the environment variable once before the loop. :)

josevalim replied:
Also, the default value of the environment variable should be set inside `def application` in the mix.exs file:

    def application do
      [env: [exclude_ecto_exceptions_from_plug: []]
    end

josevalim commented:
Thank you, i have added some comments.

alvinlindstam commented:
Thanks for the comments. I have fixed and squashed the commit.

However, I don't seem to be able to read the default env variable when declaring it in the mix file and using `fetch_env!`:

```
== Compilation error on file lib/phoenix_ecto/plug.ex ==
** (ArgumentError) application :phoenix_ecto is not loaded, or the configuration parameter :exclude_ecto_exceptions_from_plug is not set
```

josevalim commented:
@alvinlindstam ah yes, because it is compilation time. So we will need both, the env listing and the compile time default.

alvinlindstam commented:
Ok, so any other changes needed?

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
